src/openads-appnexus/service/ReplayEventBus.js

```diff
@@ -0,0 +1,93 @@
+import {observerErrorThrown} from './observerErrorThrown'
+
+class ReplayEventBus {
+  constructor() {
+    this._observers = new Map()
+    this._pendingEvents = new Map()
+  }
+
+  register({eventName, observer}) {
+    if (!eventName) {
+      throw new Error('Event Name is required')
+    }
+    if (typeof observer !== 'function') {
+      throw new Error('Observer must be a function')
+    }
+    if (!this._observers.has(eventName)) {
+      this._observers.set(eventName, [observer])
+    } else {
+      this._observers.get(eventName).push(observer)
+    }
+    this._replayPendingEvents({eventName, observer})
+  }
+
+  raise({event}) {
+    this._addPendingEvent({event})
+    if (this._observers.has(event.eventName)) {
+      this._observers
+        .get(event.eventName)
+        .forEach(observer => this._processEvent({event, observer}))
+    }
+  }
+
+  _addPendingEvent({event}) {
+    if (!this._pendingEvents.has(event.eventName)) {
+      this._pendingEvents.set(event.eventName, [event])
+    } else {
+      this._pendingEvents.get(event.eventName).push(event)
+    }
+  }
+
+  _replayPendingEvents({eventName, observer}) {
+    if (this._pendingEvents.has(eventName)) {
+      this._pendingEvents
+        .get(eventName)
+        .forEach(pendingEvent =>
+          this._processEvent({event: pendingEvent, observer})
```

Comment:
Once a pending event is processed, shouldn't it be removed from the pending list?